pkg/operator/controller/machineconfig/crio.go

```diff
@@ -32,84 +33,77 @@ import (
 	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
 )
 
-const (
-	// CrioProfilingConfigName is the name CRI-O MachineConfig CR
-	CrioProfilingConfigName = "10-crio-nodeobservability"
-
-	// CrioUnixSocketConfFile is the name of the CRI-O config file
-	CrioUnixSocketConfFile = "10-mco-profile-unix-socket.conf"
+// enableCrioProf checks if CRI-O MachineConfig CR for
+// enabling profiling exists, if not creates the resource
+func (r *MachineConfigReconciler) enableCrioProf(ctx context.Context) (bool, error) {
+	createdNow := false
+	namespace := types.NamespacedName{Name: CrioProfilingConfigName}
 
-	// CrioServiceFile is the name of the CRI-O systemd service unit
-	CrioServiceFile = "crio.service"
+	err := r.createCrioProfConf(ctx)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return createdNow, err
+	}
+	if err == nil {
+		createdNow = true
+	}
 
-	// CrioUnixSocketConfData contains the configuration required
-	// for enabling CRI-O profiling
-	CrioUnixSocketConfData = `[Service]
-Environment="ENABLE_PROFILE_UNIX_SOCKET=true"`
-)
+	// grace time for client cache to get refreshed
+	time.Sleep(500 * time.Millisecond)
 
-// ensureCrioProfConfigExists checks if CRI-O MachineConfig CR for
-// enabling profiling exists, if not creates the resource
-func (r *MachineConfigReconciler) ensureCrioProfConfigExists(ctx context.Context) (*mcv1.MachineConfig, bool, error) {
-	namespace := types.NamespacedName{Name: CrioProfilingConfigName}
-	criomc, exists, err := r.fetchCrioProfileConfig(ctx, namespace)
+	criomc, err := r.fetchCrioProfConf(ctx, namespace)
 	if err != nil {
-		return nil, false, err
+		return createdNow, fmt.Errorf("failed to ensure crio profiling config was indeed created: %v", err)
```

Comment:
nit, let's wrap those errors:
```suggestion
		return createdNow, fmt.Errorf("failed to ensure crio profiling config was indeed created: %w", err)
```